protocol/messenger_handler.go

```diff
@@ -406,20 +412,114 @@ func (m *Messenger) handleCommandMessage(state *ReceivedMessageState, message *c
 	return nil
 }
 
-func (m *Messenger) HandleBackup(state *ReceivedMessageState, message protobuf.Backup) error {
+func (m *Messenger) HandleBackup(state *ReceivedMessageState, message protobuf.Backup) []error {
+	var errors []error
+	err := m.handleSyncProfile(state, message.Profile, message.Clock)
+	if err != nil {
+		errors = append(errors, err)
+	}
+
 	for _, contact := range message.Contacts {
-		err := m.HandleSyncInstallationContact(state, *contact)
+		err = m.HandleSyncInstallationContact(state, *contact)
 		if err != nil {
-			return err
+			errors = append(errors, err)
 		}
 	}
 
 	for _, community := range message.Communities {
-		err := m.handleSyncCommunity(state, *community)
+		err = m.handleSyncCommunity(state, *community)
 		if err != nil {
-			return err
+			errors = append(errors, err)
 		}
 	}
+	err = m.handleSyncSettings(state, message.Settings)
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	if m.config.messengerSignalsHandler != nil {
+		response := wakusync.SyncFromWakuResponse{}
+
+		response.AddWakuSyncDetails(SyncWakuSectionKeyProfile, message.ProfileDetails)
```

Comment:
This does not take into account different backups, you might receive multiple and from different devices

Reply:
Your question makes me think that it's quite impossible then to have progress update in place as it's planned by UI design, cause we cannot be sure 100% ever that we're receiving the latest backed up messages. That further implies that in the middle of receiving the last known backed up messages we receive a newer one which will reset the progress as their progress details may be completely different.

So what could be a solution for that if we want to keep progress for fetching messages from waku in the UI?

Reply:
> Your question makes me think that it's quite impossible then to have progress update in place as it's planned by UI design, cause we cannot be sure 100% ever that we're receiving the latest backed up messages. 

Yes, that's correct, we can't be 100% sure, and as you pointed out, we might have to restart the process, or you might have multiple backups from multiple devices, both of them relevant, coming at different times etc.

Now, what we do about it, it's a discussion to be had, we either don't show a detailed progress (i.e only a generic spinner, which suits better our technology), or if we want to show a detailed progress, we accept the fact it will not be reliable in some cases, it will restart in others etc. Of course there might be other technical solutions, but the fact that we have permissionless non-consistent decentralized storage makes these things a nightmare to work with.

This would have been caught early if we had a tech feasibility analysis, but what we do about it at this stage is up to us.

cc @John-44 @xAlisher